### DISCUSSION TITLE
How to get nested object properties for style specification

### DISCUSSION THREAD

**Discussion by glaroc** [maplibre/maplibre-gl-js#4004] [Q&A]
Hi, I have a PMTiles file with polygon properties in a nested structure like this, 

````
properties:{"species_counts":{"id13":3,"id15":5}}
````

Now, I want to style each feature, based on the value of id13, for example. I've tried to get the nested object property with something like 

````
"fill-color": [
     "case",
        [">", ["get", "id13", ["get", "species_counts"]],0],
      "orange",
      "blue",
]
````
but anytime I try to put a nested "get" like this, nothing works. Any ideas on how to do this ?

**Comment by HarelM** [comment]
Don't use nested objects is the best advise I can give you I think...
Nested objects are not well supported in the mvt spec, they get covered to strings, which isn't very useful...

**Reply by glaroc** [reply]
OK, thanks. This seemed possible with Mapbox (https://stackoverflow.com/a/74574816/511575), so I was under the impression that it would also work with Maplibre. I'll try to unnest my data structure.

**Reply by HarelM** [reply]
I'm not sure how correct this SO question is to be honest, but I might be wrong...

**Reply by glaroc** [reply]
See also: https://stackoverflow.com/a/56569263/511575

**Reply by HarelM** [reply]
In that case, you have your answer :-)

**Reply by glaroc** [reply]
As mentioned in the question, it doesn't seem to work with Maplibre for some reason.

**Reply by mattixpet** [reply]
We tried to use nested objects, and getting the value worked with `["get", "value", ["get", "object_with_value"]]`, but indeed showed some strange behavior. We used the value to show circles on a map, which did display correctly, but clicking the feature only worked at the center.

**Comment by wipfli** [comment]
Flatten the nested data for example like this:

```json
{"species_counts":{"id13":3,"id15":5}}
```

flatten to

```json
{
  "species_counts:id13": 3,
  "species_counts:id15": 5
}
```
